x/pki/keeper/noc_root_certificates.go

```diff
@@ -70,6 +70,54 @@ func (k Keeper) RemoveNocRootCertificates(
 	))
 }
 
+func (k Keeper) RemoveNocRootCertificate(ctx sdk.Context, vid int32, subject, subjectKeyID string) {
+	certs, found := k.GetNocRootCertificates(ctx, vid)
+	if !found {
+		return
+	}
+
+	certsBefore := len(certs.Certs)
+	for i := 0; i < len(certs.Certs); {
+		cert := certs.Certs[i]
+		if cert.Subject == subject && cert.SubjectKeyId == subjectKeyID {
+			certs.Certs = append(certs.Certs[:i], certs.Certs[i+1:]...)
+		} else {
+			i++
+		}
+	}
+
+	if len(certs.Certs) == 0 {
+		k.RemoveNocRootCertificates(ctx, vid)
+	} else if certsBefore > len(certs.Certs) { // Update state only if any certificate is removed
```

Comment:
Should we add a similar check to the `RemoveNocCertificate` function?

Reply:
Will be implemented at the next PR of `Revocation of NOC non-root certificates`